### DISCUSSION TITLE
CoC change from Contributor Covenant -> GNU Kind Communications Guidelines

### DISCUSSION THREAD

**Discussion by ErikPrantare** [tenacityteam/tenacity-legacy#182] [General]
The Contributor Covenant is pretty vague in how it is written. You could read it and still not understand what behavior is not allowed, e.g. this point:

> Examples of unacceptable behavior by participants include:
> ...
> Other conduct which could reasonably be considered inappropriate in a professional setting

This has made many people vary of this specific CoC, as it could easily be bent to be used as a political tool in internal power struggles.

I propose that we use (something inspired by) the GNU Kind Communications Guidelines instead.
(https://www.gnu.org/philosophy/kind-communication.html)
This CoC is clearer in its language. It makes clear *why* we have a CoC (something not well communicated by the CC), and also gives a pretty concrete list of how one actually should act (rather than the more abstract points in the CC).
I think it does a better job of being a reference for internal culture and behavior than the CC.

**Comment by Songtech-0912** [comment]
@ErikPrantare I get the idea that the CoC may not be perfect. The Contributor Covenant may be vague, but writing anything ourselves is bound to cause even more issues IMO. We don't have the legal/team management expertise to deal with a situation in which our custom CoC causes even more issues and is even more exploited - one that is likely bound to happen because unlike the Contributor Covenant, which has been battle-tested, a custom CoC is not and introduces effectively zero-day vulnerabilities.

Instead, I think it would be a better idea to show the GNU kind communications guidelines *after* new contributors and experienced ones alike agree to the contributor covenant. After all, the GNU kind communications guidelines are *guidelines* in name after all. While it may not give us power of enforcement (like the Contributor Covenant does) it gives us every ability to discourage bad behavior by clearly pointing out and scolding people for divisive/damaging conduct. However, this is just my idea, feel free to criticize it if you wish.

**Reply by JYamihud** [reply]
Codes of Conduct are not Software licenses. They should not be taken as seriously as law. It's just a suggestion of how to behave well. I think using the original text of the GNU kind communication guidelines is enough. Or simply linking to it. This may solve a lot of issues. 

Also we can do a Lunduke way and state "Be excellent to each other." as the whole text of the new code of conduct.

**Comment by JYamihud** [comment]
Just read through it. All of it is just plain reasonable. I would implement it directly from the GNU Guidelines

**Comment by Sciss** [comment]
This is a cultural problem. The reason GNU / FSF / Stallman use this instead of a CoC is precisely mirrored by the problems relating to his resignation and following reinstallment to the FSF board, and the subsequent rms-open-letter and the rms-support-letter chasm. This proposal thus boils down to whether we take having an actionable CoC seriously, or if we prefer the church of GNU aka "free speech"…

**Reply by ErikPrantare** [reply]
I don't agree that the problems surrounding the RMS controversy stems from the GKCG (GNU Kind Communications Guidelines). Having that discussion here, however, will not lead anywhere. Let's instead try to discuss the contents of the CoC.

Yes, the way the CC is worded makes it more "actionable". That's not the primary issue I take with it. What I find problematic about it is specifically the vague language, and the fact that you're left none the wiser after having read it as to what behavior is expected of you. I think the GKCG does a better job of addressing this issue. Because what exactly is

> Other conduct which could reasonably be considered inappropriate in a professional setting

, other than a catch-all clause?

**Reply by Sciss** [reply]
I did not say the problems stem from the GKCG, but the latter is an expression of the former. A CoC is also not an algorithm, but a text that needs, like any etiquette, human judgment and context-specific interpretation. When you design an etiquette, you want to make it applicable in a future circumstance even if that has not been able to foresee. This is spirit-of-a-law vs letters-of-a-law. Other conduct reasonably considered inappropriate in a professional setting is not a catch-all, but opens the possibility to apply the spirit of the CoC to situations which may depend on the particular professional context in which it is used. That's where the reasonable comes in, it's not a blanko statement. You would have to explicitly reason to defend a decision based on "other conduct". If you look around for projects that use this CoC, this may for example be: using offensive user names, or publishing private discussions without consent, or starting to offer commercial products in a channel.

**Reply by ErikPrantare** [reply]
Let me rephrase. I don't think the issue stemmed from the FSF at all. Again though, that discussion will not be able to be resolved here.

Have we seen any case where a project with the CC has had to withstand a crisis? I don't doubt that it would be used for mostly reasonable cases in peace time, but how about withstanding pressure attacks from corporations and media conglomerates? The vagueness and "spirit-of-law"-ness of the CC makes it a prime attack vector for hostile actors. I don't see anything in it to defend against that as long as the attacker possesses all the necessary signifiers (think your typical liberal mega-corp).

**Comment by Songtech-0912** [comment]
Ok, so let's admit this is a bit of a heated topic at the very least. Allow me to introduce the following unbiased discussion of the topic (or at least as unbiased as I can make it).

First, the key phrase, *"other conduct which could reasonably be considered inappropriate in a professional setting"* in the Contributor Covenant (which I'll refer to as the **CC** from now on for convenience) stems directly from this motivation:

* The CC introduction explains that "adopting Contributor Covenant helps makes your community’s values explicit"
* It reinforces this notion with a passage that says "deal with it in a professional manner...do not respond with sarcasm, attack the code of conduct or its enforcement, or the accuser", or that essentially, these are explicit things that are **non-acceptable in a professional setting**
* Essentially, conduct that is considered "inappropriate in a professional setting" is *whatever conflicts with your community's values* plus the examples placed above, and while that may not be clear from just a quick read, it is *all but inferred* and project maintainers would understandably be able to interpret it as such

Second, the CC is, as its author puts it, an *incomplete solution*. Coraline Ehmke (its author) says this about the CC:

> "As a leader you are responsible for the safe, fair, and transparent enforcement of your community’s code of conduct. A code of conduct without such enforcement sends a false signal that a community is welcoming and inclusive, which can have a disastrous impact on marginalized or otherwise vulnerable people."

Thus, there is no conflict between the Kind Communication Guidelines and CC, even if their respective exigences are different. This is also why the CC has catch-all phrases like the "other conduct..." clause, as it is designed to fit to "the special norms and values that distinguish your own community", as explained on its overview.

Third, let's examine the key differences between the Kind Communication Guidelines and the CC (with source citations to back this up):

* Firstly, the motivation of the kind communication guidelines is given in the preamble, and is stated as such: "...people are sometimes discouraged from participating in GNU development because of certain patterns of communication that strike them as unfriendly, unwelcoming, rejecting, or harsh"
   * Thus, the _kind_ communication guidelines aims to primarily address instances of *unkind communication* in GNU/FSF projects
   * Any connection with the Stallman controversy is likely superficial in this case, even if it is correlated by coincidence
* Second, the motivation of the CC is stated as such: "[the CC] is a code of conduct that you can adapt to express both these fundamental shared values, and...signals your commitment to creating a welcoming and safe environment for everyone"
   * Thus, the CC is primarily focused on establishing community values and for keeping your community welcoming and safe
   * It is nothing without enforcement, same as the Kind Communication Guidelines, but the difference lies in that the CC is more broad and more flexible to individual communities (in terms of their specific values) while the Kind Communication Guidelines was created specifically to establish the **values of the GNU community** and thus is more detailed in terms of what it prescribes
   * To conclude, the idea is that the CC is broad because you're supposed to adjust it for your needs, while the Kind Communication Guidelines are very specific because they have 1 singular purpose and are directly made for 1 community, not for universal adoption
